### ISSUE TITLE
--noop mode not working

### ISSUE THREAD

**Issue by blackHatMonkey** [relekang/python-semantic-release#73]
I noticed this bug where the `--noop` mode is not working. This bug is present on any branch.

Steps to reproduce:
1. Edit a file
2. Add the file to git
3. Commit(make sure that it bumps the version)
4. Run `semantic-release --noop version`
5. Check logs

In the logs it can be seen that the version was bumped as and a new commit as well a new tag is created.

**Comment by relekang** [comment]
That seems like a not so nice bug, nothing was pushed tho?

**Comment by blackHatMonkey** [comment]
Nothing was pushed. It just created a new tag and also a new commit bumping the version. It seems to just ignore the option and run it normally.

**Comment by relekang** [comment]
It is used here: https://github.com/relekang/python-semantic-release/blob/master/semantic_release/cli.py#L52

It has been a while since I have worked with click, might the problem be that only `semantic-release version --noop` is supported and not `semantic-release --noop version`?

**Comment by blackHatMonkey** [comment]
The output definitely different:
```
$ semantic-release --noop version
Creating new version..
Current version: 1.0.2
Bumping with a minor version to 1.1.0.
```
As it does not include should have bumped from phrase at all.

**Comment by blackHatMonkey** [comment]
Just tried it `semantic-release version --noop` and it works as it produces the following:
```
Creating new version..
Current version: 1.0.2
No operation mode. Should have bumped from 1.0.2 to 1.1.0.
```
But it's weird since the documentation indicates option should come fist
```
Usage: semantic-release [OPTIONS] COMMAND
```

**Comment by relekang** [comment]
Yeah, options should preferably work on both sides of the command.

**Comment by relekang** [comment]
I have a hard time reproducing this:

```
$ python -m semantic_release.cli --noop version --patch                                                                                                                                                                                   rolf@snape
Creating new version..
Current version: 3.11.0
No operation mode. Should have bumped from 3.11.0 to 3.11.1.

$ python -m semantic_release.cli version --noop --patch                                                                                                                                                                                   rolf@snape
Creating new version..
Current version: 3.11.0
No operation mode. Should have bumped from 3.11.0 to 3.11.1.
```

**Comment by pareyesv** [comment]
I just noticed the bug is still active:

```
$ cat src/__init__.py
__version__ = '0.7.1'
$ semantic-release version --noop
Creating new version..
Current version: 0.7.1
No operation mode. Should have bumped from 0.7.1 to 0.8.0.
$ semantic-release --noop version
Creating new version..
Current version: 0.7.1
Bumping with a minor version to 0.8.0.
$ cat src/__init__.py
__version__ = '0.8.0'
$ python -m pip freeze | grep semantic
python-semantic-release==3.11.2
```

**Comment by weishang** [comment]
I think the bug is still there. 

semantic-release --noop version
Creating new version..
Current version: 1.4.0
Bumping with a patch version to 1.4.1.

but 

semantic-release  version --noop
Creating new version..
Current version: 1.4.0
No operation mode. Should have bumped from 1.4.0 to 1.4.1.

Currently using version python-semantic-release==4.3.4

Thank you for the tool, btw, very helpful.

**Comment by relekang** [comment]
This is weird.
```shell
python -m semantic_release.cli --noop version # working as expected
python -m semantic_release.cli version --noop # working as expected
semantic_release --noop version # still releasing
semantic_release version --noop # working as expected
```

It seems like the argument is set correctly:
```
➤ DEBUG=* semantic-release --noop version
  semantic_release:cli 'main args:' {'noop': True, 'force_level': None, 'post': False, 'retry': False, 'define': ()} +0.0us
  semantic_release:cli 'main env:' 'PYPI_USERNAME="None",PYPI_PASSWORD="None",GH_TOKEN="None",GL_TOKEN="None",' +82.0us
  semantic_release:cli 'main config:' {'check_build_status': 'false', 'commit_message': 'Automatically generated by python-semantic-release', 'commit_parser': 'semantic_release.history.angular_parser', 'patch_without_tag': 'false', 'upload_to_pypi': 'true', 'version_source': 'commit'} +74.75us
```

**Comment by bilalshaikh42** [comment]
The reason you are seeing different behavior for those two cases is because of the argument sorting that is being done in the cli.py 
If you print the __name__ variable in the two cases, you will see that it is __main__ when using the python -m syntax and semantic_release.cli when running directly. Changing the if statement to run in both these cases fixes the issue

**Comment by bilalshaikh42** [comment]
While that solution did fix the issue, it caused problems with the test runner. Instead, I added an `entry` method that does the same thing and changed `setup.py` to use that method as the entry point. 
I have added this fix to #166
